### PR TITLE
sqlsrv and pdo_sqlsrv; use Microsoft ODBC 18 when available, support Debian Bullseye

### LINKED ISSUE #432: Missing MSSQL support for Debian Bullseye/11
see https://github.com/mvorisek/image-php/actions/runs/1243314446 and https://github.com/mvorisek/image-php/commit/0868d7263af21537661db5a326e0450cd41e7050 source files/changes.

This repo should test againt Debian Buster & Bullseye images.

### PR DESCRIPTION
**Pull request by mlocati**
Close #432

### CODE REVIEW

**Comment by mvorisek** [comment]
This fix seems to no fully replace my previous fix, see https://github.com/mvorisek/image-php/commit/73116867fc873f465354b930aea9537e4d25d880

CI https://github.com/atk4/data/runs/5258494124
started failing after using the image with the commit above applied
https://github.com/atk4/data/runs/5259022227

MSSQL nor PHP DBAL lib unchanged

exception trace:
```
1) Atk4\Data\Tests\ConditionSqlTest::testBasic
Doctrine\DBAL\Exception\DriverException: An exception occurred in the driver: SQLSTATE[08001]: [Microsoft][ODBC Driver 18 for SQL Server]SSL Provider: [error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:self signed certificate]

/__w/data/data/vendor/doctrine/dbal/src/Driver/API/SQLSrv/ExceptionConverter.php:67
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:1815
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:1758
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:344
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:1532
/__w/data/data/src/Persistence/Sql/Connection.php:237
/__w/data/data/src/Persistence/Sql/Connection.php:193
/__w/data/data/src/Persistence/Sql.php:80
/__w/data/data/src/Persistence.php:55
/__w/data/data/src/Schema/TestCase.php:36
phpvfscomposer:///__w/data/data/vendor/phpunit/phpunit/phpunit:97

Caused by
Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[08001]: [Microsoft][ODBC Driver 18 for SQL Server]SSL Provider: [error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:self signed certificate]

/__w/data/data/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:30
/__w/data/data/vendor/doctrine/dbal/src/Driver/PDO/SQLSrv/Driver.php:48
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:342
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:1532
/__w/data/data/src/Persistence/Sql/Connection.php:237
/__w/data/data/src/Persistence/Sql/Connection.php:193
/__w/data/data/src/Persistence/Sql.php:80
/__w/data/data/src/Persistence.php:55
/__w/data/data/src/Schema/TestCase.php:36
phpvfscomposer:///__w/data/data/vendor/phpunit/phpunit/phpunit:97

Caused by
PDOException: SQLSTATE[08001]: [Microsoft][ODBC Driver 18 for SQL Server]SSL Provider: [error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:self signed certificate]

/__w/data/data/vendor/doctrine/dbal/src/Driver/PDO/SQLSrv/Driver.php:42
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:342
/__w/data/data/vendor/doctrine/dbal/src/Connection.php:1532
/__w/data/data/src/Persistence/Sql/Connection.php:237
/__w/data/data/src/Persistence/Sql/Connection.php:193
/__w/data/data/src/Persistence/Sql.php:80
/__w/data/data/src/Persistence.php:55
/__w/data/data/src/Schema/TestCase.php:36
phpvfscomposer:///__w/data/data/vendor/phpunit/phpunit/phpunit:97
```

is the ODBC lib installed really 1:1 replacement or does it miss any configuration?

File: install-php-extensions [hunk 1]
```diff
@@ -1511,7 +1515,14 @@ installMicrosoftSqlServerODBC() {
...
 			fi
 			printf -- '- installing the APT package\n'
-			DEBIAN_FRONTEND=noninteractive ACCEPT_EULA=Y apt-get -qy --no-install-recommends install '^msodbcsql[0-9]+$'
+			if test $PHP_MAJMIN_VERSION -le 703; then
+				DEBIAN_FRONTEND=noninteractive ACCEPT_EULA=Y apt-get -qy --no-install-recommends install msodbcsql17
+			elif test $DISTRO_VERSION_NUMBER -ge 9 && $DISTRO_VERSION_NUMBER -le 10; then
+				# On Debian 9 and 10 we have both msodbcsql17 and msodbcsql18: let's install just one
+				DEBIAN_FRONTEND=noninteractive ACCEPT_EULA=Y apt-get -qy --no-install-recommends install msodbcsql18
+			else
+				DEBIAN_FRONTEND=noninteractive ACCEPT_EULA=Y apt-get -qy --no-install-recommends install '^msodbcsql[0-9]+$'
```
Comment (mvorisek) [inline install-php-extensions:1524]: This else still installs both versions on Debian 18.

https://github.com/mvorisek/image-php/commit/6bdcd5df242ba999685322f5e196db40dfef55b1

there should be probably no regex and only `msodbcsql18` until new major release of Debian and MSSQL ODBC is out

Comment (mlocati) [reply install-php-extensions:1524]: Debian 18?

Comment (mvorisek) [reply install-php-extensions:1524]: Debian 11.2, a typo

**Pull request merged**